### ISSUE TITLE
As a user, I want to be able to create multiple sequences of action stages

### ISSUE THREAD

**Issue by othyn** [othyn/macos-auto-clicker#7]
Very far off and blue sky thinking, but I think it would be cool to allow for multiple action stages to be created and then ordered.

So the UI/UX would be a draggable text interface, something like:

---

**Action Stages**

= Every `<duration field>` press `<input field>` `<amount of actions field>` times and wait `<duration field>` before doing it again for a total of `<amount of iterations field>` times.

= Every `<duration field>` press `<input field>` `<amount of actions field>` times and wait `<duration field>` before doing it again for a total of `<amount of iterations field>` times.

---

In which the user could re-arrange the sequence based on their preferred order of execution.

Taking this further, you could even do conditional logic, but I'm not sure what conditions would be useful or if the feature would be at all.

**Comment by james-s-w-clark** [comment]
Sounds great! 
If it's any help, I find this autoclicker useful on Windows - https://github.com/federicadomani/AutoClicker2-Record-Play-The-Lists-Of-Mouse-Clicks

The extended mode lets you use global hotkeys to build up a list of clicks at different positions and with different delays (see the right-hand-side of https://github.com/federicadomani/AutoClicker2-Record-Play-The-Lists-Of-Mouse-Clicks/raw/master/screenshots_new/v5.9.3.1/AutoClicker2_v5.9.3.2.jpg?raw=true). I'd love to have similar for MacOS.

**Comment by othyn** [comment]
Ah, that looks like an excellent point of reference! I haven't thought about having the user specify/record their mouse actions or movement yet. I'll add that as another issue.

**Comment by dabroz** [comment]
This would be cool for idle games, eg. to be able press the left mouse button 10 times a second + press the 'H' key every 10 seconds.

**Comment by othyn** [comment]
Indeed, thats the intention. Just have to find the time to completely re-write the core of the application and UI 😂
